org.planqk.atlas.core/src/main/java/org/planqk/atlas/core/services/BackendServiceImpl.java

```
@@ -0,0 +1,128 @@
+        // set to list conversion is faster as lists may have duplicate objects
+        List<Backend> persistedList = new ArrayList<>(persistedSet);
+        List<Backend> listToUpdate = new ArrayList<>(setToUpdate);
+        return updateSome(persistedList, listToUpdate);
+    }
+
+    /**
+     * Unfortunately, repo.findAll() only returns lists this is the reason why this
+     * method exists in the first place. This method will create new Backends not yet
+     * saved.
+     * @param persistedSet current list of Backends
+     * @param setToUpdate Backends to update
+     * @return updated Set
+     */
+    private Set<Backend> updateSome(List<Backend> persistedSet, List<Backend> setToUpdate) {
+        // quicker, but kinda ugly way to do this is by sorting the backends
+        // and using a single for loop to compare them
+        Set<Backend> result = new HashSet<>();
+        boolean foundCurrentBackend = false;
+        for (Backend backend: setToUpdate) {
+            for (Backend persistedBackend: persistedSet) {
+                // found the same backend
+                if (backend.getId() == persistedBackend.getId()) {
+                    // did it change?
+                    if (!backend.equals(persistedBackend)) {
+                        update(persistedBackend.getId(), backend);
+                    }
+                    foundCurrentBackend = true;
+                }
+            }
+            // the current backend is new, save it
+            if (!foundCurrentBackend) {
+                result.add(save(backend));
+            }
+        }
+        return result;
+    }
+
+    @Override
+    public void delete(UUID id) {
+        // only delete if unused in SoftwarePlatforms and CloudServices
+        long count = cloudServiceRepository.countCloudServiceByBackend(id) + softwarePlatformRepository.countSoftwarePlatformByBackend(id);
+        if (count == 0) {
```

should this be a no-op otherwise?

Maybe throw an exception to let the user know?